pkg/app/piped/executor/ecs/ecs.go

```diff
@@ -0,0 +1,149 @@
+		return &deployExecutor{
+			Input: in,
+		}
+	}
+	r.Register(model.StageECSSync, f)
+
+	r.RegisterRollback(model.ApplicationKind_ECS, func(in executor.Input) executor.Executor {
+		return &rollbackExecutor{
+			Input: in,
+		}
+	})
+}
+
+func findCloudProvider(in *executor.Input) (name string, cfg *config.CloudProviderECSConfig, found bool) {
+	name = in.Application.CloudProvider
+	if name == "" {
+		in.LogPersister.Errorf("Missing the CloudProvider name in the application configuration")
+		return
+	}
+
+	cp, ok := in.PipedConfig.FindCloudProvider(name, model.CloudProviderECS)
+	if !ok {
+		in.LogPersister.Errorf("The specified cloud provider %q was not found in piped configuration", name)
+		return
+	}
+
+	cfg = cp.ECSConfig
+	found = true
+	return
+}
+
+func loadServiceDefinition(in *executor.Input, serviceDefinitionFile string, ds *deploysource.DeploySource) (types.Service, bool) {
+	in.LogPersister.Infof("Loading service manifest at the %s commit (%s)", ds.RevisionName, ds.RevisionName)
+
+	serviceDefinition, err := provider.LoadServiceDefinition(ds.AppDir, serviceDefinitionFile)
+	if err != nil {
+		in.LogPersister.Errorf("Failed to load ECS service definition (%v)", err)
+		return types.Service{}, false
+	}
+
+	in.LogPersister.Infof("Successfully loaded the ECS service definition at the %s commit", ds.RevisionName)
+	return serviceDefinition, true
+}
+
+func loadTaskDefinition(in *executor.Input, taskDefinitionFile string, ds *deploysource.DeploySource) (types.TaskDefinition, bool) {
+	in.LogPersister.Infof("Loading service manifest at the %s commit (%s)", ds.RevisionName, ds.RevisionName)
+
+	taskDefinition, err := provider.LoadTaskDefinition(ds.AppDir, taskDefinitionFile)
+	if err != nil {
+		in.LogPersister.Errorf("Failed to load ECS service definition (%v)", err)
+		return types.TaskDefinition{}, false
+	}
+
+	in.LogPersister.Infof("Successfully loaded the ECS service definition at the %s commit", ds.RevisionName)
+	return taskDefinition, true
+}
+
+func sync(ctx context.Context, in *executor.Input, cloudProviderName string, cloudProviderCfg *config.CloudProviderECSConfig, taskDefinition types.TaskDefinition, serviceDefinition types.Service) bool {
+	in.LogPersister.Infof("Start applying the ECS task definition")
+	client, err := provider.DefaultRegistry().Client(cloudProviderName, cloudProviderCfg, in.Logger)
+	if err != nil {
+		in.LogPersister.Errorf("Unable to create ECS client for the provider %s: %v", cloudProviderName, err)
+		return false
+	}
+
+	// Build and publish new version of ECS service and task definition.
+	ok := build(ctx, in, client, taskDefinition, serviceDefinition)
+	if !ok {
+		in.LogPersister.Errorf("Failed to build new version for ECS %s", serviceDefinition.ServiceName)
+		return false
+	}
+
+	in.LogPersister.Infof("Successfully applied the service definition and the task definition for ECS service %s and task definition %s", serviceDefinition.ServiceName, taskDefinition.TaskDefinitionArn)
```

Comment:
`serviceDefinition.ServiceName` and `taskDefinition.TaskDefinitionArn` are both pointer to string not string itself will lead to wrong string format like this `Successfully applied the service definition and the task definition for ECS service %!s(*string=0xc000c00cd0) and task definition %!s(*string=0xc000a89a30)` , please use value instead 👀